sql/core/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala

```diff
@@ -52,7 +52,7 @@ private[sql] object AvroUtils extends Logging {
       spark: SparkSession,
       options: Map[String, String],
       files: Seq[FileStatus]): Option[StructType] = {
-    val conf = spark.sessionState.newHadoopConfWithOptions(options)
+    val conf = spark.sessionState.asInstanceOf[SessionState].newHadoopConfWithOptions(options)
```

Comment:
Changes like this are not very reasonable. It's only needed to work around the current IntelliJ limitation of not recognizing the dependency exclusion.

I'm a bit hesitant to merge this PR. If we need a temporary fix for IntelliJ support, shall we use maven profiles and the default profile does not include `connect-shim`?

Reply:
This can be experimentally tested in Intellij by manually ignoring the `connect-shims` module. We will observe that this action causes the `sql-api` module to be marked with compilation errors. For instance, the definitions of `JavaRDD` and `RDD` cannot be found when working with classes like `org.apache.spark.sql.DataFrameReader`.